### PR TITLE
v1.8 backports 2020-09-25

### PR DESCRIPTION
**Pull request by vadorovsky**
* #13241 -- cilium: encrypt-node creates two IPsec tunnels but only uses one (@jrfastab)
 * ~#13228 -- identity: Avoid kvstore lookup for local identities (@gandro)~ too many unit tests fail, I will try in an another PR
 * #13248 -- hubble: Fix filter by reply reporting flows with unknown reply state  (@gandro)
 * #13187 -- docs: backport documentation additions (@kkourt)
 * #13234 -- EKS: improve rules for asymmetric routing (multi-node NodePort) (@qmonnet)
 * #13162 -- bpf: fix nodeport ipv6 services (@fristonio)
 * #13272 -- k8s: Remove CRD deleting functionality (@christarazi)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 13241 13248 13187 13234 13162 13272; do contrib/backporting/set-labels.py $pr done 1.8; done
```

### CODE REVIEW

**Comment by mrostecki** [comment]
test-backport-1.8

**Comment by mrostecki** [comment]
@christarazi I'm a bit unsure if I backported your PR properly. I had a conflict with https://github.com/christarazi/cilium/commit/c2ca49c4c1704ca7531deb331b1ae2af8c33b926.

**Review by qmonnet — APPROVED** [review]
> * #13234 -- EKS: improve rules for asymmetric routing (multi-node NodePort) (@qmonnet)

Looks good to me, thank you!

**Comment by aanm** [comment]
@mrostecki the code is failing:
```
13:00:33  
13:00:33  # github.com/cilium/cilium/test/k8sT
13:00:33  k8sT/Services.go:130:47: not enough arguments in call to kubectl.GetCiliumPodOnNode
13:00:33  	have (string)
13:00:33  	want (string, string)
13:00:33  
13:00:33  Ginkgo ran 1 suite in 18.309450923s
13:00:33  Test Suite Failed
```

**Comment by mrostecki** [comment]
test-backport-1.8

File: pkg/k8s/apis/cilium.io/v2/client/register.go [hunk 1]
```diff
@@ -503,11 +503,7 @@ func createUpdateCRD(clientset apiextensionsclient.Interface, CRDName string, cr
 		return false, err
 	})
 	if err != nil {
-		deleteErr := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(context.TODO(), crd.ObjectMeta.Name, metav1.DeleteOptions{})
-		if deleteErr != nil {
-			return fmt.Errorf("unable to delete k8s %s CRD %s. Deleting CRD due: %s", CRDName, deleteErr, err)
-		}
-		return err
+		return fmt.Errorf("error occurred waiting for CRD: %w", err)
```
Comment (christarazi) [inline pkg/k8s/apis/cilium.io/v2/client/register.go:506]: This is correct. 

What's missing is the removal of the code in `operator/crd_test.go` as seen in the upstream commit.

**Review by christarazi — CHANGES_REQUESTED** [review]

**Review by christarazi — APPROVED** [review]
Disregard my previous comments, I realized that `operator/crd_test.go` doesn't have any logic related to deleting CRDs. This is good to go for me!

**Review by fristonio — APPROVED** [review]
Looks Good for my changes. Thanks :rocket:

**Comment by mrostecki** [comment]
Failure is legit, let me fix it.

**Comment by mrostecki** [comment]
test-backport-1.8

**Comment by aanm** [comment]
@mrostecki I would suggest to drop the faulty commits so we can merge the other changes.

**Comment by mrostecki** [comment]
@aanm OK, let's try without #13228.

**Comment by mrostecki** [comment]
test-backport-1.8

**Comment by mrostecki** [comment]
test-backport-1.8

**Comment by mrostecki** [comment]
test-backport-1.8

**Comment by mrostecki** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Kubernetes-Upstream/2541/console - VM provisioning failed

**Comment by mrostecki** [comment]
test-backport-1.8

**Pull request merged**